resources/views/index.blade.php

```diff
@@ -516,58 +524,109 @@ class="my-prism-editor"
+                var rules = rule.split('|');
+                validations.isRequired = isRuleApplied(rules, /^required$/)
+                validations.isDate = isRuleApplied(rules, /^date$/)
+                validations.isArray = isRuleApplied(rules, /^array$/)
+                validations.isString = isRuleApplied(rules, /^string$/)
+                validations.isEmail = isRuleApplied(rules, /^email$/)
+                validations.isIn = isRuleApplied(rules, /^in:(.+)$/)
+                validations.isRequiredIf = isRuleApplied(rules, /^required_if:(.+)$/)
+
+                if (isRuleApplied(rules, /^integer$/)) {
                     validations.isInteger = true
                 }
-                if (rule.match(/min:([0-9]+)/)) {
-                    validations.min = rule.match(/min:([0-9]+)/)[1]
+
+                if (isRuleApplied(rules, /^min:([0-9]+)$/)) {
+                    validations.min = getAppliedRule(rules, /^min:([0-9]+)$/)[1]
                     if (!validations.min) {
                         validations.min = 1
                     }
                 }
-                if (rule.match(/max:([0-9]+)/)) {
-                    validations.max = rule.match(/max:([0-9]+)/)[1]
+
+                if (isRuleApplied(rules, /^max:([0-9]+)$/)) {
+                    validations.max = getAppliedRule(rules, /^max:([0-9]+)$/)[1]
                     if (!validations.max) {
                         validations.max = 100
                     }
                 }
+
+                if (validations.isIn) {
+                    validations.in = (getAppliedRule(rules, /^in:(.+)$/)[1]).split(',')
+                }
+
+                if (validations.isRequiredIf) {
+                    validations.required_if = (getAppliedRule(rules, /^required_if:(.+)$/)[1]).split(',')
+                }
             })
 
             if (validations.isString) {
                 validations.value = faker.name.findName()
             }
+
             if (validations.isInteger) {
                 validations.value = Math.floor(Math.random() * (validations.max - validations.min + 1) + validations.min)
             }
+
             if (validations.isDate) {
                 validations.value = new Date(faker.date.between(new Date(), new Date())).toISOString().slice(0, 10)
             }
 
+            if (validations.isEmail) {
+                validations.value = faker.internet.email().toLowerCase()
+            }
+
+            if (validations.isIn) {
+                validations.value = faker.random.arrayElement(validations.in)
+            }
+
+            if (validations.isRequiredIf) {
+                if (validations.required_if.length !== 2) {
+                    throw "Required if needs 2 parameters"
+                    return;
+                }
+
+                var srcAttribute = validations.required_if[0]
+                if (!source.hasOwnProperty(srcAttribute)) {
+                    source[srcAttribute] = guessValue(srcAttribute, rules, source);
```

Comment:
There are two issues with this pull req.

1. This recursion is too risky. It leads to `RangeError: Maximum call stack size exceeded`.
2. After I tested it, the POST parameters are ok and the newly added refresh works fine. However, the auto generated parameters for GET requests are gone.